### PR TITLE
Various code cleanup.

### PR DESCRIPTION
**Pull request by millems**

### CODE REVIEW

File: core/src/main/java/software/amazon/awssdk/SdkStandardLoggers.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import software.amazon.awssdk.annotation.SdkInternalApi;
+import software.amazon.awssdk.utils.Logger;
+
+/**
+ * A centralized set of loggers that used across the SDK to log particular types of events. SDK users can then specifically enable
+ * just these loggers to get the type of that they want instead of having to enable all logging.
+ */
+@SdkInternalApi
+public class SdkStandardLoggers {
```
Comment (shorea) [inline core/src/main/java/software/amazon/awssdk/SdkStandardLoggers.java]: +1 I like this.

File: core/src/main/java/software/amazon/awssdk/http/InterruptMonitor.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    private InterruptMonitor() {}
+
+    /**
+     * Check if the thread has been interrupted. If so throw an {@link InterruptedException}.
+     * Long running tasks should be periodically checked if the current thread has been
+     * interrupted and handle it appropriately
+     *
+     * @throws InterruptedException If thread has been interrupted
+     */
+    // TODO address
```
Comment (shorea) [inline core/src/main/java/software/amazon/awssdk/http/InterruptMonitor.java]: I added the address mainly to find a better home for it. So you can remove this unless there's something else you wanted to do with it.

File: core/src/main/java/software/amazon/awssdk/http/pipeline/stages/HandleResponseStage.java [hunk 3]
```diff
@@ -152,16 +150,4 @@ private void closeInputStreamIfNeeded(HttpResponse httpResponse,
...
-    /**
-     * Determines whether a request failed or not based on the response. If the response is an empty optional then
-     * execution failed with some non-service error like an IOException or some other client error. If the response
-     * is a fulfilled optional and {@link Response#isFailure()} is true then the execution failed with a service error
-     * of some kind.
-     *
-     * @param response Optional of unmarshalled response.
-     * @return True if the response was a failure. False if it was a success.
-     */
-    private boolean didRequestFail(Response<OutputT> response) {
```
Comment (shorea) [inline core/src/main/java/software/amazon/awssdk/http/pipeline/stages/HandleResponseStage.java]: Lol nice. I'd like to think at one point it did more than this.

File: pom.xml [hunk 4]
```diff
@@ -280,6 +281,12 @@
                 <version>${assertj.version}</version>
             </dependency>
             <dependency>
+                <groupId>log4j</groupId>
+                <artifactId>log4j</artifactId>
+                <version>1.2.17</version>
+                <scope>${log4j.version}</scope>
```
Comment (shorea) [inline pom.xml]: Should this be here? Scope looks wrong.

File: core/src/main/java/software/amazon/awssdk/http/pipeline/stages/HandleResponseStage.java (hunk 3 above)
Comment (millems) [reply core/src/main/java/software/amazon/awssdk/http/pipeline/stages/HandleResponseStage.java]: It did!

File: pom.xml (hunk 4 above)
Comment (millems) [reply pom.xml]: whoops!

**Pull request merged**